File: python/simplify-path.py
class Solution:
    # Time complexity: O(n) where n the length of the input string is
    # Space complexity: O(n)
    def simplifyPath(self, path: str) -> str:
        path = path.split("/")
        res = []
        
        for p in path:
            if not p or p == ".": continue
            if p == "..":
                if res:
                    res.pop()
            else:
                res.append(p)
                
        return "/" + "/".join(res)

File: js/simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
// Time complexity: O(n) where n the length of path is
// Space complexity: O(n)
var simplifyPath = function(path) {
    path = path.split("/")
    const res = []
    
    for (p of path) {
        if (p === "" || p === ".") continue
        if (p == "..") {
            if (res.length > 0) res.pop()
        } else {
            res.push(p)
        }
    }
    return "/" + res.join("/")
};